dissect/target/plugins/os/windows/locale.py

```diff
@@ -0,0 +1,66 @@
+from dissect.target.exceptions import (
+    RegistryKeyNotFoundError,
+    RegistryValueNotFoundError,
+)
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export
+
+WindowsKeyboardRecord = TargetRecordDescriptor(
+    "windows/keyboard",
+    [
+        ("string", "layout"),
+        ("string", "id"),
+    ],
+)
+
+
+class LocalePlugin(Plugin):
+    def __init__(self, target):
+        super().__init__(target)
+        self.LANG_DICT = {
+            lang.name: lang.value
+            for k in self.target.registry.key("HKLM\\SYSTEM\\CurrentControlSet\\Control\\Keyboard Layout\\DosKeybCodes")
+            for lang in k.values()
+        }
+
+    def check_compatible(self):
+        try:
+            self.target.registry.key("HKLM\\SYSTEM\\CurrentControlSet\\Control\\Keyboard Layout\\DosKeybCodes")
+            self.target.registry.key("HKLM\\SYSTEM\\CurrentControlSet\\Control\\TimeZoneInformation")
+            assert len(self.target.registry.key("HKCU\\Keyboard Layout\\Preload").keys) > 0
+            assert len(self.target.registry.key("HKCU\\Control Panel\\International\\User Profile").keys) > 0
+        except (RegistryKeyNotFoundError, RegistryValueNotFoundError, AssertionError):
+            return False
```

Comment:
This won't actually work because `__init__` runs before `check_compatible`. Personally I'd revert it to the way it was and "assume" that these keys will always exist. We already do this in multiple places, and I don't think we have a very nice solution for it at the time. So if we ever do think of one we'll fix it everywhere at the same time.

You can however change the `check_compatible` to the following:
```suggestion
    def check_compatible(self):
        return self.target.has_function("registry")
```